Watson Training Data importer (#7935)
RasaHQ/rasa: docs/docs/migrate-from/ibm-watson-to-rasa.mdx

description: Open source alternative to IBM Watson for conversational bots and NLP
---

## Step 1 Export your data from Watson

Follow the <a className="reference external" href="https://developer.ibm.com/tutorials/learn-how-to-export-import-a-watson-assistant-workspace/" target="_blank">instructions</a> and export your bot's data.

## Step 2: Create a Rasa Project

To create a Rasa project, run:

```bash
rasa init
```

This will create a directory called `data`.
Remove the files in this directory, and
move your json file with Watson data into this directory.

```bash
rm -r data/*
mv /path/to/file.json data/
```

## Step 3: Train your NLU model

To train a model using your Watson data, run:

```bash
rasa train nlu
```

## Step 4: Test your NLU model

Let's see how your NLU model will interpret some test messages.
To start a testing session, run:

```bash
rasa shell nlu
```

This will prompt your for input.
Type a test message and press 'Enter'.
The output of your NLU model will be printed to the screen.
You can keep entering messages and test as many as you like.

> Press 'control + C' to quit.

Reviewer: we need to adapt this, right?

description: Open source alternative to IBM Watson for conversational bots and NLP
---

## Step 1

> To use <a className="reference external" href="https://developer.ibm.com/tutorials/learn-how-to-export-import-a-watson-assistant-workspace/" target="_blank">exported IBM Watson workspaces</a>

Reviewer 2: ```suggestion
Follow the <a className="reference external" href="https://developer.ibm.com/tutorials/learn-how-to-export-import-a-watson-assistant-workspace/" target="_blank">instructions</a> and export your bot's data.
```

description: Open source alternative to IBM Watson for conversational bots and NLP
---

> ## Step 1 Export your data from Watson

Reviewer: @melindaloubser1 Can you please double check the docs?

> Follow the <a className="reference external" href="https://developer.ibm.com/tutorials/learn-how-to-export-import-a-watson-assistant-workspace/" target="_blank">instructions</a> and export your bot's data.

Reviewer 3: ```suggestion
Follow the <a className="reference external" href="https://developer.ibm.com/tutorials/learn-how-to-export-import-a-watson-assistant-workspace/" target="_blank">instructions</a> to export your bot's data.
```

## Step 2: Create a Rasa Project

To create a Rasa project, run:

```bash
rasa init
```

This will create a directory called `data`.
Remove the files in this directory, and

> move your json file with Watson data into this directory.

Reviewer 3: ```suggestion
move the json file containing the exported Watson data into this directory.
```

Reviewer 3: See below - may be best to leave out moving the file

```bash
rm -r data/*

> mv /path/to/file.json data/

Reviewer 3: does watson give you a file called `file.json`? If not here's a suggested rename for clarity:
```suggestion
mv /path/to/exported_watson_file.json data/
```

```

> ## Step 3: Train your NLU model

Reviewer 3: ```suggestion
## Step 3: Convert your Watson NLU data
```

> To train a model using your Watson data, you should convert the data to a format Rasa can understand.:

Reviewer 3: ```suggestion
To convert your Watson data to a format Rasa can understand, run:
```

```bash

> rasa data convert -i data/your-watson-file.json -o data/file.yml --format yaml

Reviewer 3: Won't they end up with the json + yaml sitting next to each other under `data`, and Rasa will try to read both? If so probably best to skip the data moving step in step (2) and change this suggestion:
```suggestion
rasa data convert -i path-your-watson-file.json -o data/converted-watson-data.yml --format yaml
```

```

## Step 4: Train your NLU model

> To train a model using your Watson data, run:

Reviewer 3: ```suggestion
To train an NLU model using your Watson data, run:
```

```bash
rasa train nlu
```

## Step 5: Test your NLU model

Let's see how your NLU model will interpret some test messages.
To start a testing session, run:

```bash
rasa shell nlu
```

This will prompt your for input.
Type a test message and press 'Enter'.
The output of your NLU model will be printed to the screen.
You can keep entering messages and test as many as you like.

> Press 'control + C' to quit.

Reviewer 3: Best to link to the CLI docs for `rasa shell`
